sympy/sets/fancysets.py

```diff
@@ -544,27 +561,32 @@ def __new__(cls, *args):
     Either the start or end value of the Range must be finite.'''))
 
         if start.is_infinite:
-            if step*(stop - start) < 0:
-                start = stop = S.One
+            if Lt(step*(stop - start), 0) == True:
+                start = end = S.Zero
+                step = S.One
             else:
                 end = stop
-        if not start.is_infinite:
+        else:
             ref = start if start.is_finite else stop
             n = ceiling((stop - ref)/step)
-            if n <= 0:
+            if (n <= 0) == True:
```

Comment:
Maybe:
```
    if n.is_extended_nonnegative:
```
I'm not sure though. Some people seem to prefer the `a < b == True` idiom